### PR TITLE
add new resources and parameters to substitute deprecate ones

### PR DESCRIPTION
**Pull request by jcalderin**
S3 configuration use to be all in one resource. Terraform/aws have splitter in different resources.
This change adds all the new resources and new V2 functions to use the new ones

Once all resources are migrated to V2, they can be renamed/removed

extra: added new parameters to alb and db_instance

### CODE REVIEW

File: tools/gen-providers.ts [hunk 1]
```diff
@@ -499,6 +500,27 @@ const sse_rule: RecordDecl = {
   ],
 };
 
+const and: RecordDecl = {
```
Comment (bacek) [inline tools/gen-providers.ts]: Those are usually prefixed with enclosed type. Something like `aws_s3_bucket_lifecycle_configuration_filter_and`. Same for `filter`.

Comment (bacek) [reply tools/gen-providers.ts]: And `generateParams` for it.

**Review by bacek — CHANGES_REQUESTED** [review]

File: tools/gen-providers.ts (hunk 1 above)
Comment (jcalderin) [reply tools/gen-providers.ts]: Done

File: library/aws/aws.ts [hunk 2]
```diff
@@ -124,6 +124,44 @@ export function createS3Bucket(
   return AR.createS3Bucket(tfgen, name, params);
 }
 
+// CreateS3Bucket for newer versions of terraform
+export function createS3BucketV2(
```
Comment (bacek) [inline library/aws/aws.ts]: Bit too many positional args. Can you move them into sub-struct?

**Review by bacek — COMMENTED** [review]
Almost there.

File: library/aws/aws.ts (hunk 2 above)
Comment (jcalderin) [reply library/aws/aws.ts]: done :)

**Review by bacek — APPROVED** [review]
LGTM++

**Pull request merged**